libraries/usersearch/impl/src/main/kotlin/io/element/android/libraries/usersearch/impl/MatrixUserRepository.kt

```diff
@@ -31,24 +32,31 @@ import javax.inject.Inject
 
 @ContributesBinding(SessionScope::class)
 class MatrixUserRepository @Inject constructor(
+    client: MatrixClient,
     private val dataSource: UserListDataSource
 ) : UserRepository {
 
+    private val sessionId = client.sessionId
+
     override suspend fun search(query: String): Flow<List<UserSearchResult>> = flow {
-        // Manually add a fake result with the matrixId, if any
-        val isUserId = MatrixPatterns.isUserId(query)
-        if (isUserId) {
+        // Manually add a fake result with the matrixId, if provided and not the local user
+        val isAnotherUsersId =  query != sessionId.value && MatrixPatterns.isUserId(query)
```

Comment:
`isAnotherUsersId` -> `isAnotherUserId`, and please format the file :)

Reply:
It was meant to read like "is another user's id", but I can see the ambiguity without punctuation :laughing:. I don't think `isAnotherUserId` quite conveys what it means, so I've changed it entirely to `shouldQueryProfile`.